.github/workflows/cd-manager.yml

```diff
@@ -0,0 +1,121 @@
+name: Publish Docker image
+
+on:
+  push:
+    branches: [ develop, release-candidate, main ]
+    paths:
+      - 'cd/manager/**'
+  pull_request: # pull requests
+  workflow_dispatch: # manually triggered
+
+env:
+  # Dagger
+  DAGGER_PLAN: ../../ci/cd-manager.cue
+  DAGGER_VERSION: "0.2.21"
+  DAGGER_LOG_FORMAT: "plain"
+  PIPELINE_TOOLS_VER: "0.1.3"
+  # Secrets
+  AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+  GH_TOKEN: ${{ secrets.GH_TOKEN }}
+  # GitHub
+  RUN_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+  BRANCH: "develop"
+  #${{ github.ref_name }}
+  SHA: ${{ github.sha }}
+
+jobs:
+  image:
+    name: Verify and publish Docker image
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Check out the repo
+        uses: actions/checkout@v2
+      -
+        name: Set sha tag
+        run: |
+          SHA_TAG=$(git rev-parse --short=12 "${{ env.SHA }}")
+          echo "SHA_TAG=$SHA_TAG" >> $GITHUB_ENV
+      -
+        name: Set main branch tag
+        if: ${{ env.BRANCH == 'main' }}
+        run: |
+          echo "ENV_TAG=prod" >> $GITHUB_ENV
+      -
+        name: Set rc branch tag
+        if: ${{ env.BRANCH == 'release-candidate' }}
+        run: |
+          echo "ENV_TAG=tnet" >> $GITHUB_ENV
+      -
+        name: Set develop branch tag
+        if: ${{ env.BRANCH == 'develop' || env.ENV_TAG == '' }}
+        run: |
+          echo "ENV_TAG=dev" >> $GITHUB_ENV
+      -
+        name: Set commit status "pending"
```

Comment:
In next iteration I'd love to make these into reusable gh components

Reply:
Yes, absolutely, good idea. We're already using this in several places now. I could maybe even add them as globally reusable steps in this repo, then use it in Ceramic, CAS, IPFS.